### PR TITLE
Unil monty component

### PR DESCRIPTION
**Pull request by lfaucon**
Not to be merged yet 

## Done
- UI which allow to create several solution with either the type Tree or Table
- Table component using code form [react-spreadsheet-component](https://github.com/felixrieseberg/React-Spreadsheet-Component)

## TODO
- Make the table collaborative. Now it keeps its own states and does not update if the props change
- Add tree view
- Add streaming and adapt ac-classifier

### CODE REVIEW

**Comment by houshuang** [comment]
Looks good. I would make the tabs and + sign bigger and bolder. Maybe it's not clear enough, maybe the + sign should be replaced with some bold text "essayer un autre fois" or something like that. Also, is "Valider" the right word choice or should it be something like "Submit"? Or even "save"? When you click Valider, what should happen? Should it go to a new tab letting you submit again or just wait until you click +?

PS: You forgot to add jquery to package.json. I pushed an update.

**Comment by lfaucon** [comment]
@houshuang I did only use a bunch of TextInputs for the tree, we could make it look good by adding a background picture. Tell me what you think.

I have a problem with the connection to sharedb in the stream method call
The code is 
```JavaScript
Meteor.methods({
  'stream': (activity, path, value) => {
    if (activity.streamTarget) {
      console.log('bonjour', activity.streamTarget)
      const docId = activity.streamTarget + '/all';
      if (Cache[docId]) {
        console.log('use cache')
        const [doc, dataFn] = Cache[docId];
        safelyInsertObject(doc, dataFn, path, value);
      } else {
        console.log('create cache')
        const doc = serverConnection.get('rz', docId);
        doc.fetch();
        doc.on('load', () => {
          console.log('create cache now')
          const dataFn = generateReactiveFn(doc);
          Cache[docId] = [doc, dataFn];
          safelyInsertObject(doc, dataFn, path, value);
        });
      }
    }
  }
});
```

I observe that the connection with the sharedb collection does not work if I created the session(activity?) before starting the server

Step 1: I start the server, then start the session. The stream method call tries to connect and create a cache, but hangs on `doc.on.('load',...)`. The stream does not work.

```
I20171012-21:53:28.823(2)? bonjour cj8ovg7mf0003x9s65fjvjw2g
I20171012-21:53:28.824(2)? create cache
I20171012-21:53:29.386(2)? bonjour cj8ovg7mf0003x9s65fjvjw2g
I20171012-21:53:29.387(2)? create cache
I20171012-21:53:29.738(2)? bonjour cj8ovg7mf0003x9s65fjvjw2g
I20171012-21:53:29.739(2)? create cache
```

STEP 2: I make any change on the code to get the server to restart. The stream starts to work fine.


```
...
I20171012-21:54:57.569(2)? Running shareDB server
=> Meteor server restarted
I20171012-21:55:01.097(2)? bonjour cj8ovg7mf0003x9s65fjvjw2g
I20171012-21:55:01.098(2)? create cache
I20171012-21:55:01.105(2)? create cache now
I20171012-21:55:07.435(2)? bonjour cj8ovg7mf0003x9s65fjvjw2g
I20171012-21:55:07.435(2)? use cache
I20171012-21:55:10.826(2)? bonjour cj8ovg7mf0003x9s65fjvjw2g
I20171012-21:55:10.827(2)? use cache
...
```


STEP 3: I restart the session. The stream again does not work.


```
I20171012-21:56:30.023(2)? bonjour cj8ovxa5y00039ws6otl0lh1n
I20171012-21:56:30.024(2)? create cache
I20171012-21:56:30.745(2)? bonjour cj8ovxa5y00039ws6otl0lh1n
I20171012-21:56:30.746(2)? create cache
I20171012-21:56:31.016(2)? bonjour cj8ovxa5y00039ws6otl0lh1n
I20171012-21:56:31.017(2)? create cache
```

I hope you have an idea of what is causing that bug :)

**Comment by lfaucon** [comment]
@houshuang It should look ok for the treeView now.

I was stuck on spinner when running a real session. I think it is due to the last changes you made to sharedb. Should be easily fixed I think

File: frog/server/share-db-manager.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ export const serverConnection = backend.connect();
 
 export const startShareDB = () => {
   if (!Meteor.settings.dont_start_sharedb) {
-    new WebSocket.Server({ server }).on('connection', ws => {
+    new WebSocket.Server({ server }).once('connection', ws => {
```
Comment (lfaucon) [inline frog/server/share-db-manager.js]: I removed this change and things seem to work

**Pull request merged**